packages/fxa-settings/src/components/Settings/Page2faSetup/index.tsx

```diff
@@ -127,6 +129,10 @@ const Page2faSetup = (_: RouteComponentProps) => {
   }, [totpInfoLoading, totpInfoError, totpInfo, showGenericError, goHome]);
 
   /* ───── early return states ───── */
+  if (!session.verified) {
```

Comment:
Guards the page in case it is reached via direct url with an unverified session (vs via the settings `add` button which also has a guard)

Reply:
This can be done in a polish PR, but it might worth noting in the comment that without this check, the `totpInfoLoading` could get stuck in a true state and we would end up with a spinner.

Reply:
Thanks, added a comment!